src/uu/cp/src/copydir.rs

```diff
@@ -452,25 +463,10 @@ pub(crate) fn copy_directory(
         }
     }
 
-    // Handle final directory permission fixes.
-    // This is almost the same as the permission-fixing code above,
-    // with minor differences (commented)
-    if let Some(last_iter) = last_iter {
-        let diff = last_iter.path().strip_prefix(root).unwrap();
-
-        // Do _not_ skip `.` this time, since we know we're done.
-        // This is where we fix the permissions of the top-level
-        // directory we just copied.
```

Comment:
The permissions for `.` _are_ preserved with the new code, right?

Reply:
yeah, it is :)